README.md

```diff
@@ -129,3 +129,53 @@ You can find your Chrome profile directory names by visiting:
 > When using both Firefox and Firefox Developer Edition on MacOS, ensure each app uses its appropriate profile.
 > For example, the "My Firefox Developer Edition Profile" in `about:profiles` should always be opened with the Firefox Developer Edition app.
 > Using the wrong profile with either app will cause them to crash.
+
+## Continuous Integration (CI) 🛠️
+
+Browsir uses GitHub Actions for continuous integration and automated workflows. The CI setup includes:
+
+### Automated Checks
+- **Commit Lint**: Enforces conventional commit messages (e.g., `feat:`, `fix:`, `docs:`, etc.)
+- **Golangci-lint**: Runs static code analysis and style checks
+- **Go Tests**: Executes all tests in the project
+- **Ad-hoc Checks**: 
+  - Checks for large files (>5MB)
+  - Scans for potential sensitive data in code files
+  - More to come
+
+### Automated PR Management
+- **Auto Rebase**: Automatically rebases PRs against the target branch
+- **Auto Merge**: Automatically merges PRs when they meet the following criteria:
+  - PR is not a draft
+  - PR is mergeable
+  - PR has the `ready-to-merge` label
+  - All CI checks pass
+
+### How to Use the CI Features
+
+1. **Commit Messages**: Follow the conventional commit format:
+   ```
+   type(scope): description
+   ```
+   Where `type` can be:
+   - `feat`: New feature
+   - `fix`: Bug fix
+   - `docs`: Documentation changes
+   - `style`: Code style changes
+   - `refactor`: Code refactoring
+   - `perf`: Performance improvements
+   - `test`: Adding or modifying tests
+   - `build`: Build system changes
+   - `ci`: CI configuration changes
+   - `chore`: Maintenance tasks
+   - `revert`: Reverting changes
+
+2. **Auto Merge**: To enable automatic merging of your PR:
+   - Ensure all CI checks pass
+   - Add the `ready-to-merge` label to your PR
+   - The PR will be automatically rebased and merged if all conditions are met
+
+### CI Configuration Files
+- `.github/workflows/ci.yaml`: Main CI workflow with linting and testing
+- `.github/workflows/auto-merge.yaml`: Automated PR management workflow
+- `.commitlintrc.json`: Commit message linting configuration
```

Comment:
After the changes requested above most of this will not be true anymore, I think you can remove it or rephrase it by add a contributor section in the readme or a separated new file like CONTRIBUTOR.md